web-marketplace/src/pages/BuyCredits/BuyCredits.utils.tsx

```diff
@@ -27,6 +41,137 @@ type GetFormModelParams = {
   projectId: string;
 };
 
+export const findMatchingSellOrders = (
+  data: BuyCreditsSchemaTypes,
+  sellOrders: UISellOrderInfo[] | undefined,
+) => {
+  if (!sellOrders) return [];
+
+  const sellOrderIds = data?.sellOrders?.map(order => order.sellOrderId);
+
+  return sellOrders.filter(order => sellOrderIds?.includes(order.id));
+};
+
+export const getOrderedSellOrders = (
+  isCard: boolean,
+  cardSellOrders: CardSellOrder[],
+  filteredCryptoSellOrders: UISellOrderInfo[],
+): UISellOrderInfo[] => {
+  return isCard
+    ? cardSellOrders.sort((a, b) => a.usdPrice - b.usdPrice)
+    : filteredCryptoSellOrders?.sort(
+        (a, b) => Number(a.askAmount) - Number(b.askAmount),
+      ) || [];
+};
+
+export const getWarningModalContent = (
+  currency: { askDenom: string; askBaseDenom: string } | undefined,
+  isWeb2UserWithoutWallet: boolean,
+  creditsInRequestedSellOrders: number,
+  _: TranslatorType,
+  allowedDenomsData: QueryAllowedDenomsResponse | undefined,
+  data: BuyCreditsSchemaTypes,
+  creditsInAllSellOrders: number | undefined,
+  isVisitingUser: boolean,
+): BuyWarningModalContent | undefined => {
+  if (
+    currency?.askDenom === USD_DENOM &&
+    (isWeb2UserWithoutWallet || isVisitingUser) &&
```

Comment:
I believe this is because of this check, I don't think we should limit this to visiting or web2 users without wallet

Reply:
Do you think we should remove line 79 altogether?